### ISSUE TITLE
Expected type hint "Datetime"; found "DateTime"

### ISSUE THREAD

**Issue by BowFarmer** [squizlabs/PHP_CodeSniffer#3817]
**Describe the bug**
I received this warning from running phpcs --standard=WordPress on a file.
Expected type hint "Datetime"; found "DateTime" for $day_to_check

**Code sample**
```php
class Foo {
    private function dbtn_get_issue_number_from_date( DateTime $day_to_check, int $start_or_end ) {
    }
}
```

**Custom ruleset** [Not sure where the problem ruleset is. I'm using the WordPress Coding Standards]
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <description>phpcodesniffer4wordpress</description>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
PHPCS output here
---------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
---------------------------------------------------------------------------------------
 408 | ERROR | Expected type hint "Datetime"; found "DateTime" for $day_to_check
---------------------------------------------------------------------------------------
```

**Expected behavior**
I expect that the hint ( DateTime $day_to_check ) has to be DateTime and not Datetime.

**Versions (please complete the following information):**
 - OS: MacOS 13.3.1 (a) (22E772610a)
 - PHP: 8.2.5
 - PHPCS: version 3.7.2 (stable) by Squiz (http://www.squiz.net)
 - Standard: Squiz

**Additional context**

**Comment by jrfnl** [comment]
@BowFarmer This report contains too little information to reproduce the issue.

I suspect the error is the `Squiz.Commenting.FunctionComment.IncorrectTypeHint` error, but that sniff only kicks in when there is a docblock and the code sample provided does not contain a docblock, making the issue impossible to reproduce.

You can see the sniff code associated with the error message by running `phpcs -s test.php --standard=WordPress`

If the above sniff is the issue, I suggest you fix up your docblock.

If not, please report back with enough information to reproduce the issue if the error code starts with `Generic`, `MySource`, `PEAR`, `PSR1`, `PSR2`, `PSR12`, `Squiz` or `Zend`.

If the error code starts with something else, please report it to whatever external standard the error is coming from.

**Comment by BowFarmer** [comment]
Hi Juliette,

Thank you for your quick reply. This is what I get when I run phpcs -s --standard=WordPress on the file.

Expected type hint "Datetime"; found "DateTime" for $day_to_check
(Squiz.Commenting.FunctionComment.IncorrectTypeHint)

On May 4, 2023, at 08:31, Juliette ***@***.***> wrote:

@BowFarmer <https://github.com/BowFarmer> This report contains too little information to reproduce the issue.
I suspect the error is the Squiz.Commenting.FunctionComment.IncorrectTypeHint error, but that sniff only kicks in when there is a docblock and the code sample provided does not contain a docblock, making the issue impossible to reproduce.

You can see the sniff code associated with the error message by running phpcs -s test.php --standard=WordPress

If the above sniff is the issue, I suggest you fix up your docblock.

If not, please report back with enough information to reproduce the issue if the error code starts with Generic, MySource, PEAR, PSR1, PSR2, PSR12, Squiz or Zend.

If the error code starts with something else, please report it to whatever external standard the error is coming from.

**Comment by jrfnl** [comment]
Closing for lack of response.
If this is still an issue, please provide the additional information requested and if the issue can then be reproduced, the ticket can be reopened.
